### DISCUSSION TITLE
Calldata compression

### DISCUSSION THREAD

**Discussion by boyuan-chen** [bobanetwork/boba_legacy#173] [Ideas]
## Calldata compression
Based on the analysis of last 100K blocks, `Brotli` is a best option for us. The estimated gas saving for the sequencer is about 48%.

```
{
    "Zlib Estimated Fee Savings": 0.2413901362319999,
    "Zlib Dictionary Estimated Fee Savings": 0.22177344951776967,
    "Brotli Estimated Fee Savings": 0.48050498924934604,
    "Total TXs": 99360,
  }
```

**Comment by souradeep-das** [comment]
Is there a significant trade-off in speed when using Brotli over other options? Although, the last 100k blocks shows the cost savings are significantly more
How do we relay the compression savings to individual txs and their costs
